// posthog-js/lib/src/extensions/extension-bundles.js
"use strict";
/**
 * Pre-grouped extension bundles for tree-shaking support.
 *
 * Use these with `__extensionClasses` to control which extensions are included in your bundle.
 * The default `posthog-js` entrypoint includes all extensions. When using `posthog-js/slim`,
 * you can import only the bundles you need:
 *
 * @example
 * ```ts
 * import posthog from 'posthog-js/slim'
 * import { ReplayExtensions, AnalyticsExtensions } from 'posthog-js/extensions'
 *
 * posthog.init('ph_key', {
 *   __extensionClasses: {
 *     ...ReplayExtensions,
 *     ...AnalyticsExtensions,
 *   }
 * })
 * ```
 *
 * @module
 */
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllExtensions = exports.TracingExtensions = exports.SiteAppsExtensions = exports.ProductToursExtensions = exports.ErrorTrackingExtensions = exports.AnalyticsExtensions = exports.SessionReplayExtensions = void 0;
var autocapture_1 = require("../autocapture");
var dead_clicks_autocapture_1 = require("./dead-clicks-autocapture");
var exception_autocapture_1 = require("./exception-autocapture");
var history_autocapture_1 = require("./history-autocapture");
var tracing_headers_1 = require("./tracing-headers");
var web_vitals_1 = require("./web-vitals");
var session_recording_1 = require("./replay/session-recording");
var heatmaps_1 = require("../heatmaps");
var posthog_product_tours_1 = require("../posthog-product-tours");
var site_apps_1 = require("../site-apps");
/** Session replay and related extensions. */
exports.SessionReplayExtensions = {
    sessionRecording: session_recording_1.SessionRecording,
};
/** Autocapture, click tracking, heatmaps, and web vitals. */
exports.AnalyticsExtensions = {
    autocapture: autocapture_1.Autocapture,
    historyAutocapture: history_autocapture_1.HistoryAutocapture,
    heatmaps: heatmaps_1.Heatmaps,
    deadClicksAutocapture: dead_clicks_autocapture_1.DeadClicksAutocapture,
    webVitalsAutocapture: web_vitals_1.WebVitalsAutocapture,
};
/** Automatic exception and error capture. */
exports.ErrorTrackingExtensions = {
    exceptionObserver: exception_autocapture_1.ExceptionObserver,
};
/** In-app product tours. */
exports.ProductToursExtensions = {
    productTours: posthog_product_tours_1.PostHogProductTours,
};
/** Site apps support. */
exports.SiteAppsExtensions = {
    siteApps: site_apps_1.SiteApps,
};
/** Distributed tracing header injection. */
exports.TracingExtensions = {
    tracingHeaders: tracing_headers_1.TracingHeaders,
};
/** All extensions — equivalent to the default `posthog-js` bundle. */
exports.AllExtensions = __assign(__assign(__assign(__assign(__assign(__assign({}, exports.SessionReplayExtensions), exports.AnalyticsExtensions), exports.ErrorTrackingExtensions), exports.ProductToursExtensions), exports.SiteAppsExtensions), exports.TracingExtensions);
